File: MB-Secretaria/review_py/review_flags.py
from __future__ import annotations

import os
from collections.abc import Mapping

from .types import HumanReviewFlags


def _read_bool(env: Mapping[str, str], name: str, fallback: bool) -> bool:
    value = env.get(name)
    if value is None:
        return fallback
    return value.strip().lower() == "true"


def load_human_review_flags(env: Mapping[str, str] | None = None) -> HumanReviewFlags:
    source = env or os.environ
    return HumanReviewFlags(
        enabled=_read_bool(source, "HUMAN_REVIEW_ENABLED", False),
        shadow_mode=_read_bool(source, "HUMAN_REVIEW_SHADOW_MODE", True),
        blocking_enabled=_read_bool(source, "HUMAN_REVIEW_BLOCKING_ENABLED", False),
        image_required=_read_bool(source, "HUMAN_REVIEW_IMAGE_REQUIRED", True),
        low_confidence_required=_read_bool(source, "HUMAN_REVIEW_LOW_CONFIDENCE_REQUIRED", True),
        multimodal_required=_read_bool(source, "HUMAN_REVIEW_MULTIMODAL_REQUIRED", True),
        high_risk_required=_read_bool(source, "HUMAN_REVIEW_HIGH_RISK_REQUIRED", True),
        override_enabled=_read_bool(source, "HUMAN_OVERRIDE_ENABLED", False),
    )

File: MB-Chat/review/review-flags.ts
import type { HumanReviewFlags } from "./types";

export const DEFAULT_HUMAN_REVIEW_FLAGS: HumanReviewFlags = {
  enabled: false,
  shadowMode: true,
  blockingEnabled: false,
  imageRequired: true,
  lowConfidenceRequired: true,
  multimodalRequired: true,
  highRiskRequired: true,
  overrideEnabled: false,
};

function readBoolean(value: string | undefined, fallback: boolean): boolean {
  if (value === undefined) return fallback;
  return value.trim().toLowerCase() === "true";
}

export function loadHumanReviewFlags(env: NodeJS.ProcessEnv = process.env): HumanReviewFlags {
  return {
    enabled: readBoolean(env.HUMAN_REVIEW_ENABLED, DEFAULT_HUMAN_REVIEW_FLAGS.enabled),
    shadowMode: readBoolean(env.HUMAN_REVIEW_SHADOW_MODE, DEFAULT_HUMAN_REVIEW_FLAGS.shadowMode),
    blockingEnabled: readBoolean(env.HUMAN_REVIEW_BLOCKING_ENABLED, DEFAULT_HUMAN_REVIEW_FLAGS.blockingEnabled),
    imageRequired: readBoolean(env.HUMAN_REVIEW_IMAGE_REQUIRED, DEFAULT_HUMAN_REVIEW_FLAGS.imageRequired),
    lowConfidenceRequired: readBoolean(
      env.HUMAN_REVIEW_LOW_CONFIDENCE_REQUIRED,
      DEFAULT_HUMAN_REVIEW_FLAGS.lowConfidenceRequired,
    ),
    multimodalRequired: readBoolean(
      env.HUMAN_REVIEW_MULTIMODAL_REQUIRED,
      DEFAULT_HUMAN_REVIEW_FLAGS.multimodalRequired,
    ),
    highRiskRequired: readBoolean(env.HUMAN_REVIEW_HIGH_RISK_REQUIRED, DEFAULT_HUMAN_REVIEW_FLAGS.highRiskRequired),
    overrideEnabled: readBoolean(env.HUMAN_OVERRIDE_ENABLED, DEFAULT_HUMAN_REVIEW_FLAGS.overrideEnabled),
  };
}
